internal/domain/usecases/document_usecase.go

```diff
@@ -29,8 +32,81 @@ func NewDocumentUsecase(cfg config.Config, logger *zap.Logger, documentRepositor
 	}
 }
 
-func (u *documentUsecase) GetAllDocuments() (*[]dtos.DocumentDTO, *apperror.AppError) {
-	return nil, nil
+func (u *documentUsecase) GetAllDocuments(req *dtos.GetAllDocumentsDTO) (*dtos.PaginationResponse, *apperror.AppError) {
+	// validate arguments
+	if org := strings.ToLower(req.Organization); org != "" &&
+		org != strings.ToLower(constant.SCCU) &&
+		org != strings.ToLower(constant.SGCU) {
+
+		u.logger.Named("GetAllDocuments").Error(constant.ErrInvalidOrg, zap.String("organization", org))
+		return nil, apperror.BadRequestError(constant.ErrInvalidOrg)
+	}
+
+	if dt := strings.ToLower(req.DocumentType); dt != "" &&
+		dt != strings.ToLower(constant.ANNOUNCEMENT) &&
+		dt != strings.ToLower(constant.STATISTIC) &&
+		dt != strings.ToLower(constant.BUDGET) {
+
+		u.logger.Named("GetAllDocuments").Error(constant.ErrInvalidDocType, zap.String("document_type", dt))
+		return nil, apperror.BadRequestError(constant.ErrInvalidDocType)
+	}
```

Comment:
I'm not sure about this, but can we do the validation in handler layer so that we can reject the request before it enters usecase layer? Also, can we `strings.ToUpper()` instead so that we don't have to `strings.ToLower()` the constants

Reply:
@wiraphatys what do you think? Where should data validation go? handler or usecase

Reply:
Also, if validations e.g. from strings to enums (e.g. document types) is used more than once in other requests in the future, let's convert it into some kind of utililty function

Reply:
@TeeGoood data validation should be in handler layer kub, cuz usecase layer is pure business logic kub bro